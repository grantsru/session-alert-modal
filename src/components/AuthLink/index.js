import React from "react";
import PropTypes from "prop-types";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";

function AuthLink({ loginHref }) {
  return (
    <React.Fragment>
      <Typography
        variant={"subheading"}
        style={{ marginBottom: "15px" }}
      >
        Your session has expired.<br />
        To login, click the button below.
      </Typography>
      <Button
        color={"default"}
        variant={"contained"}
        href={loginHref}
      >Login</Button>
    </React.Fragment>
  );
}

AuthLink.propTypes = {
  loginHref: PropTypes.string.isRequired
};

export default AuthLink;
