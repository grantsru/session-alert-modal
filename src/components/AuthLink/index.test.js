import React from "react";
import AuthLink from "./index";
import { create } from "react-test-renderer";

const login = "foo";
const handleButtonClick = () => null;

describe("AuthLink", () => {
  test("it matches the snapshot", () => {
    const component = create(
      <AuthLink
        login={login}
        click={handleButtonClick}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
