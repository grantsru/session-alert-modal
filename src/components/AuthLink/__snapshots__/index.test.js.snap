// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLink it matches the snapshot 1`] = `
Array [
  <h3
    className="MuiTypography-root-1 MuiTypography-subheading-8"
    style={
      Object {
        "marginBottom": "15px",
      }
    }
  >
    Your session has expired. To login, click the button below.
  </h3>,
  <a
    className="MuiButtonBase-root-63 MuiButton-root-37 MuiButton-contained-48 MuiButton-raised-51"
    href="foo"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="MuiButton-label-38"
    >
      Login
    </span>
    <span
      className="MuiTouchRipple-root-66"
    />
  </a>,
]
`;
